### PR TITLE
feat(transactions): send_7702_tx

### LINKED ISSUE #128: Add EIP-7702 example: `send_eip7702_transaction.rs`
EIP-7702 adds a new tx type (`SET_CODE_TX`) which contains a new list: `authorization_list`. Each item in this list is a tuple of `(chain_id, address, nonce)` which is signed. Since this is an additional signing flow, it might be a bit hard to figure out how to use it.

One important distinction is that the items in the `authorization_list` are the actual accounts affected: i.e., the accounts "abstracted" in this AA proposal are the signers of each individual item. This means that the signatory of the main transaction is actually a sponsor.

It is possible a nice example here requires some additional helpers in Alloy for working with authorization lists.

See the EIP for more details: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7702.md

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes #128 

## Solution

send 7702 tx. Need latest anvil and alloy

## PR Checklist

- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: examples/transactions/examples/send_eip7702_transaction.rs
Comment (mattsse) [inline examples/transactions/examples/send_eip7702_transaction.rs:85]: can use assert_eq!

File: examples/transactions/examples/send_eip7702_transaction.rs
Comment (mattsse) [inline examples/transactions/examples/send_eip7702_transaction.rs:86]: same

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic nit, can merge after new alloy release

**Review by zerosnacks — APPROVED** [review]
PR lgtm, I don't think we would want to merge this into main with the patch and instead wait for the next Alloy release?

File: Cargo.toml
Comment (zerosnacks) [inline Cargo.toml:122]: TODO: remove

**Review by onbjerg — APPROVED** [review]

File: examples/transactions/examples/send_eip7702_transaction.rs
Comment (DaniPopes) [inline examples/transactions/examples/send_eip7702_transaction.rs:17]: can you add the command you used to compile this

**Comment by yash-atreya** [comment]
Awaiting new alloy release before merging

**Comment by onbjerg** [comment]
@yash-atreya alloy 0.3.0 is released, so this should be unblocked

**Pull request merged**
